Sources/SnapshotAssertion/SnapshotAssertion.swift

```diff
@@ -1,90 +1,71 @@
 import XCTest
 
-@discardableResult
-public func assertSnapshot(
-  matching view: UIView,
-  identifier: String? = nil,
-  _ file: StaticString = #file,
-  _ function: String = #function,
-  _ line: UInt = #line)
-  -> XCTAttachment? {
-
-    let fileURL = URL(fileURLWithPath: String(describing: file))
-    let snapshotsDirectoryURL = fileURL.deletingLastPathComponent()
-      .appendingPathComponent("__Snapshots__")
-    let fileManager = FileManager.default
-
-    try! fileManager.createDirectory(
-      at: snapshotsDirectoryURL,
-      withIntermediateDirectories: true,
-      attributes: nil
-    )
-
-    UIGraphicsBeginImageContextWithOptions(view.bounds.size, false, 0)
-    let context = UIGraphicsGetCurrentContext()!
-    view.layer.render(in: context)
-    let image = UIGraphicsGetImageFromCurrentImageContext()!
-    UIGraphicsEndImageContext()
-    let data = UIImagePNGRepresentation(image)!
+var diffTool: String? = nil
 
-    let snapshotName = fileURL.deletingPathExtension().lastPathComponent
-      + "_\(function.prefix(function.count - 2))"
-      + (identifier.map({ "_" + $0 }) ?? "")
-      + ".png"
-
-    let snapshotURL = URL(string: snapshotName, relativeTo: snapshotsDirectoryURL)!
+public protocol Diffable {
+  static var diffableFileExtension: String? { get }
+  var diffableData: Data { get }
+  func diff(comparing other: Data) -> XCTAttachment?
+}
 
-    guard fileManager.fileExists(atPath: snapshotURL.path) else {
-      try! data.write(to: snapshotURL)
-      return nil
-    }
+public protocol Snapshot {
```

Comment:
This protocol describes types that can have snapshots taken, deferring to a `Diffable` type.